packages/common/src/types/semanticLayer.ts

```diff
@@ -20,3 +28,21 @@ export type SemanticLayerQuery = {
     dimensions: string[];
     metrics: string[];
 };
+
+export interface SemanticLayerTransformer<
+    ViewType,
+    QueryType,
+    DimensionsType,
+    MetricsType,
+    ResultsType,
+    SqlType,
+> {
+    fieldsToSemanticLayerFields: (
+        dimensions: DimensionsType,
+        metrics: MetricsType,
+    ) => SemanticLayerField[];
+    viewsToSemanticLayerViews: (views: ViewType[]) => SemanticLayerView[];
+    semanticLayerQueryToQuery: (query: SemanticLayerQuery) => QueryType;
+    resultsToResultRows: (results: ResultsType) => Record<string, any>[];
```

Comment:
Checking what we can do here 🤔

Reply:
we have this in the FE

```
export type ResultValue = {
    raw: unknown;
    formatted: string;
};

export type ResultRow = Record<string, { value: ResultValue }>;
```

Reply:
We ended up deciding to use a new type so that FE takes care of formatting and we don't have to handle `null` values

```
export type SemanticLayerResultRow = Record<
    string,
    string | number | boolean | null
>;
```